flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -1264,4 +1271,76 @@ public void run() {
 			}
 		}
 	}
+
+	// ------------------------------------------------------------------------
+	//  metrics
+	// ------------------------------------------------------------------------
+
+	private class InputBuffersGauge implements Gauge<Integer> {
```

Comment:
I had pull these classes out into TaskIOMetricGroup since the buffers metrics is Task I/O scope